### PR TITLE
Update priorities in inspect form on category change.

### PR DESCRIPTION
**Pull request by dracos**
Different categories may have a different list of priorities, so store them all and update as the category changes.

### CODE REVIEW

File: web/cobrands/fixmystreet/fixmystreet.js [hunk 1]
```diff
@@ -573,6 +573,16 @@ $.extend(fixmystreet.set_up, {
...
+        // And update the associated priority list
+        var priorities = $("form#report_inspect_form " + selector).data('priorities');
+        var $select = $('#problem_priority'),
+            curr_pri = $select.val();
+        $select.find('option:gt(0)').remove();
+        $.each(priorities.split('&'), function(i, kv) {
+            kv = kv.split('=', 2);
+            $select.append($('<option>', { value: kv[0], text: decodeURIComponent(kv[1]) }));
+        });
+        $select.val(curr_pri);
```
Comment (davea) [inline web/cobrands/fixmystreet/fixmystreet.js]: If there are no priorities found, this sets the label of the control to 'undefined':
screen shot 2016-12-15 at 15 20 43

File: perllib/FixMyStreet/App/Controller/Report.pm [hunk 2]
```diff
@@ -312,6 +312,26 @@ sub inspect : Private {
...
+            foreach (@{$c->stash->{categories}}) {
+                push @{$priorities_by_category->{$_}}, $pri->id . '=' . URI::Escape::uri_escape_utf8($pri->name);
+            }
+        }
+    }
+    foreach (keys %{$priorities_by_category}) {
+        $priorities_by_category->{$_} = join('&', @{$priorities_by_category->{$_}});
+    }
+
+    $c->stash->{priorities_by_category} = $priorities_by_category;
```
Comment (davea) [inline perllib/FixMyStreet/App/Controller/Report.pm]: There's some deduplication required here, as it's possible for the same priority to appear multiple times (e.g. `<p data-category="Drainage" class="hidden" data-priorities="2=Cat%201A&2=Cat%201A&2=Cat%201A&3=Cat%201B&4=Cat%204">`):
screen shot 2016-12-15 at 15 21 45

File: perllib/FixMyStreet/App/Controller/Report.pm [hunk 3]
```diff
@@ -312,6 +312,26 @@ sub inspect : Private {
     $c->stash->{categories} = $c->forward('/admin/categories_for_point');
     $c->stash->{report_meta} = { map { $_->{name} => $_ } @{ $c->stash->{problem}->get_extra_fields() } };
 
+    my @priorities = $c->model('DB::ResponsePriority')->for_bodies($problem->bodies_str_ids)->all;
```
Comment (davea) [inline perllib/FixMyStreet/App/Controller/Report.pm]: I think this is causing some two-tier weirdness with the wrong priorities being shown if you change to a category in a different council:
priorities
Selecting the priorities by the report's location, not its `bodies_str`, would probably fix this.

File: web/cobrands/fixmystreet/fixmystreet.js (hunk 1 above)
Comment (dracos) [inline web/cobrands/fixmystreet/fixmystreet.js]: Empty string splitting to one element array gets me every time!

File: perllib/FixMyStreet/App/Controller/Report.pm (hunk 3 above)
Comment (dracos) [inline perllib/FixMyStreet/App/Controller/Report.pm]: Hmm, yes, it should only add the priority that is on all categories, to those categories in the same body as the priority.

**Pull request merged**